packages/primevue/src/treetable/BodyCell.vue

```diff
@@ -29,7 +29,8 @@
                 :data-p-partialchecked="partialChecked"
             >
                 <template #icon="slotProps">
-                    <component v-if="templates['checkboxicon']" :is="templates['checkboxicon']" :checked="slotProps.checked" :partialChecked="partialChecked" :class="slotProps.class" />
+                    <!-- TODO: It should be decided which slot to use for controlling the checkbox: checkboxicon, rowcheckboxicon -->
```

Comment:
- The Treetable documentation states that there are three slots for controlling the checkbox icon: checkboxicon, rowcheckboxicon, and headercheckboxicon. 
- In contrast, the DataTable, which is similar to Treetable, provides rowcheckboxicon and headercheckboxicon.
- Therefore, it needs to be decided which of the following two cases aligns with the implementation purpose for the Treetable.

Case 1: If the purpose is to use checkboxicon, the explanations for rowcheckboxicon and headercheckboxicon should be removed from the documentation.

Case 2: If the intent is to use rowcheckboxicon and headercheckboxicon like the DataTable, then checkboxicon should be removed from the documentation, and the code should be modified as proposed in the PR. (PR: To ensure existing functionality, it has been modified to guarantee the use of two slots (checkboxicon, rowcheckboxicon).